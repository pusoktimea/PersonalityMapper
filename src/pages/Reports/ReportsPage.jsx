import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import Loader from 'components/Loader';
import Panel from 'components/Panel';
import Button from 'components/Button';
import Table from 'components/Table';
import TableColumn from 'components/Table/TableColumn';

import './reports-page.scss';

class ReportsPage extends PureComponent {
  static propTypes = {
    isReportsLoaded: PropTypes.bool.isRequired,
    onGetReports: PropTypes.func.isRequired,
    reports: PropTypes.object.isRequired,
    isSideBarMinimised: PropTypes.bool.isRequired
  }

  componentWillMount() {
    if (!this.props.isReportsLoaded) {
      this.props.onGetReports();
    }
  }

  render() {
    const {
      reports: {
        list
      },
      isReportsLoaded,
      isSideBarMinimised
    } = this.props;

    const listWithVendorReport = [
      ...list,
      {
        name: 'Vendor Report',
        isVendorReport: true
      }
    ];
    const baseClass = 'main-content';

    return (
      <div className={cx('reports-page', baseClass, isSideBarMinimised && `${baseClass}--stretched`)}>
        <h2 className="title">Reports</h2>
        {
          !isReportsLoaded ?
            <Loader /> :
            <Panel>
              <Table
                bordered={false}
                hovered={false}
                striped={false}
                selectable={false}
                items={listWithVendorReport}
              >
                <TableColumn
                  header="Report Name"
                  cellClassName="report-name-cell"
                  contentGetter={(item) => (
                    <Button
                      href={item.isVendorReport ? '/vendor_report' : `/reports/${item.id}`}
                    >
                      {item.name}
                    </Button>
                  )}
                />
              </Table>
            </Panel>
        }
      </div>
    );
  }
}

export default ReportsPage;
