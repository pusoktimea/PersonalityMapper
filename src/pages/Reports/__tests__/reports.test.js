import React from 'react';
import renderer from 'react-test-renderer';
import {StaticRouter} from 'react-router';

import ReportsPage from '../ReportsPage';

const pageProps = {
  isReportsLoaded: true,
  onGetReports: jest.fn(),
  reports: {
    list: [{
      name: 'rep 1',
      id: 'rep_1'
    }, {
      name: 'rep 2',
      id: 'rep_2'
    }]
  },
  isSideBarMinimised: false
};

describe('Reports Page test', () => {
  it('renders and matches snapshot', () => {
    expect(
      renderer.create(
        <StaticRouter context={{}}>
          <ReportsPage {...pageProps} />
        </StaticRouter>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
