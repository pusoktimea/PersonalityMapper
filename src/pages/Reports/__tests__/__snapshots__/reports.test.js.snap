// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports Page test renders and matches snapshot 1`] = `
<div
  className="reports-page main-content"
>
  <h2
    className="title"
  >
    Reports
  </h2>
  <div
    className="activate-panel"
  >
    <div
      className="activate-panel_body"
    >
      <table
        className="activate-table activate-table--condensed"
      >
        <thead
          className="activate-table_header"
        >
          <tr>
            <th
              className="activate-table_header_item"
            >
              Report Name
            </th>
          </tr>
        </thead>
        <tbody
          className="activate-table_body"
        >
          <tr
            className="activate-table_row activate-table_row--odd"
            onClick={[Function]}
          >
            <td
              className="activate-table_row_column report-name-cell"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default"
                disabled={false}
                href="/reports/rep_1"
                onClick={[Function]}
              >
                rep 1
              </a>
            </td>
          </tr>
          <tr
            className="activate-table_row"
            onClick={[Function]}
          >
            <td
              className="activate-table_row_column report-name-cell"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default"
                disabled={false}
                href="/reports/rep_2"
                onClick={[Function]}
              >
                rep 2
              </a>
            </td>
          </tr>
          <tr
            className="activate-table_row activate-table_row--odd"
            onClick={[Function]}
          >
            <td
              className="activate-table_row_column report-name-cell"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default"
                disabled={false}
                href="/vendor_report"
                onClick={[Function]}
              >
                Vendor Report
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
